sonic-xcvrd/xcvrd/xcvrd_utilities/y_cable_helper.py

```diff
@@ -58,6 +100,87 @@ def _wrapper_get_presence(physical_port):
     return y_cable_platform_sfputil.get_presence(physical_port)
 
 
+def y_cable_wrapper_get_transceiver_info(physical_port):
```

Comment:
@vdahiya12 I got a better idea of hooking the simulated y_cable driver.
For the simulated y_cable driver to work, we need to inject a configuration file like `/etc/sonic/mux_simulator.json`. The config file needs to supply information like IP/port of the mux simulator.

So, we can decorate this function like this:
```suggestion
def hook_y_cable_simulated(target):
    """Decorator to add hook for using the simulated y_cable driver.

    This decorator checks existence of the configuration file required by the simulated y_cable driver. If the
    configuration file is found, then override the "manufacturer" and "model" fields with value "microsoft" and 
    "simulated" in the collected transceiver info dict. Consequently, instance of the simulated y_cable driver 
    class will be initialized.

    When the configuration file is not found on system, then just return the original transceiver info to initialize
    instance of y_cable driver class of whatever actually plugged physical y_cable.

    For test systems using simulated y_cable, we can just inject the simulated y_cable driver config file then
    restart the pmon service before testing starts.

    Args:
        target (function): The function collecting transceiver info.
    """    
    MUX_SIMULATOR_CONFIG_FILE = "/etc/sonic/mux_simulator.json"
    VENDOR = "microsoft"
    MODEL = "simulated"
    def wrapper(*args, **kwargs):
        res = target(*args, **kwargs)
        if os.path.exists(MUX_SIMULATOR_CONFIG_FILE):
            res["manufacturer"] = VENDOR
            res["model"] = 
        return res
    wrapper.__name__ = target.__name__
    return wrapper


@hook_y_cable_simulated
def y_cable_wrapper_get_transceiver_info(physical_port):
```

When config file of simulated y_cable driver is found on system, then update the "manufacturer" and "model" in the collected transceiver info dict to "microsoft" and "simulated.

Reply:
added to code, fixed

Reply:
added in xcvrd, fixed